### PR TITLE
[MJAR-292] - Detect MRJAR and add Multi-Release manifest entry

### PR DESCRIPTION
**Pull request by jorsol**
https://issues.apache.org/jira/browse/MJAR-292

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [X] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MJAR) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [X] Each commit in the pull request should have a meaningful subject line and body.
 - [X] Format the pull request title like `[MJAR-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MJAR-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [X] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [X] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [X] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [X] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [X] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java [hunk 1]
```diff
@@ -225,23 +237,24 @@ public File createArchive()
...
-            if ( includedFile.endsWith( MODULE_DESCRIPTOR_FILE_NAME ) )
-            {
-                containsModuleDescriptor = true;
-                break;
-            }
+            getLog().debug( "Adding 'Multi-Release: true' manifest entry." );
+            archive.addManifestEntry( "Multi-Release", "true" );
         }
 
         String archiverName = containsModuleDescriptor ? "mjar" : "jar";
```
Comment (slawekjaranowski) [inline src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java:260]: Should we also use `mjar` as archiver name when detect in steps above that is Multi-Release?

I don't know what is difference between `mjar` and `jar` archiver ...

File: src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java [hunk 2]
```diff
@@ -225,23 +237,24 @@ public File createArchive()
...
+        // May give false positives if the files is named as module descriptor
+        // but is not in the root of the archive or in the versioned area
+        // (and hence not actually a module descriptor).
+        // That is fine since the modular Jar archiver will gracefully
+        // handle such case.
+        // And also such case is unlikely to happen as file ending
+        // with "module-info.class" is unlikely to be included in Jar file
+        // unless it is a module descriptor.
+        boolean containsModuleDescriptor =
+            Arrays.stream( includedFiles ).anyMatch( p -> p.endsWith( MODULE_DESCRIPTOR_FILE_NAME ) );
```
Comment (slawekjaranowski) [inline src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java:251]: I would like to move this lines to pleace where is used about lines 260

File: src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java [hunk 3]
```diff
@@ -225,23 +237,24 @@ public File createArchive()
...
-            // And also such case is unlikely to happen as file ending
-            // with "module-info.class" is unlikely to be included in Jar file
-            // unless it is a module descriptor.
-            if ( includedFile.endsWith( MODULE_DESCRIPTOR_FILE_NAME ) )
-            {
-                containsModuleDescriptor = true;
-                break;
-            }
+            getLog().debug( "Adding 'Multi-Release: true' manifest entry." );
+            archive.addManifestEntry( "Multi-Release", "true" );
```
Comment (slawekjaranowski) [inline src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java:257]: What when user already defined such configuration ... do we need to discover in such case?

Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java]: I know that will be overridden by the same value but we can omit computation when we expect the same result. 

But ok, anyMatch method should finish quickly so cost should not be too much.

**Comment by jira-importer** [comment]
Resolve #443

**Pull request merged**